das/state.go

```diff
@@ -35,14 +49,26 @@ type coordinatorState struct {
 	catchUpDoneCh chan struct{}
 }
 
+// retry represents a retry attempt with a backoff delay.
+type retry struct {
+	// count specifies the number of retry attempts made so far.
+	count int
+	// after specifies the time for the next retry attempt.
+	after time.Time
```

Comment:
Can you also make a method on retry called `After()` that will do this inside: 

```
return retry.after.After(time.Now())
```
so that you only have to call if retry.After() ? looks a bit cleaner

You can even rename method to make it more readable.